dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java

```diff
@@ -621,23 +621,24 @@ private void setProcessInstance(ProcessInstance processInstance, Tenant tenant,
             throw new RuntimeException("workflow instance is null");
         }
 
+        Date scheduleTime = processInstance.getScheduleTime();
         Map<String, String> timeParams = BusinessTimeUtils
                 .getBusinessTime(processInstance.getCmdTypeIfComplement(),
-                        processInstance.getScheduleTime());
+                        scheduleTime !=null ? scheduleTime : processInstance.getStartTime());
```

Comment:
Thanks for your contribute, I think that can't be modify it like this, because system.datetime is definited for processIntance scheduleTime. When scheduleTime is null, system.datetime is reasonable to use the current time.

Reply:
Thanks, but i have a process define that has three jobs, and every job write data to mysql and use timestamp(${system.datetime}) as versioncode. sometime, i need manual execute this process to update to fix datas. after that when i want verify these three job instances have the same timestamp same as versioncode in mysql, go to view parameters page that will display current timestamp that does not same as versioncode. And this is true problem in my work, how to resolve it? maybe a processinstance should have a runtime timestamp only one?

Reply:
I think you should use other instead of ${system.datetime}, such as $[yyyyMMddHHmmss]

Reply:
$[yyyyMMddHHmmss] will have different for three jobs, because the three jobs have front and back dependencies. I expect a consistent timestamp as the versioncode

Reply:
I think this need other way, if such a change is made here, it will have a great impact